ios/brave-ios/Sources/PlaylistUI/MediaContentView.swift

```diff
@@ -0,0 +1,287 @@
+          toggleFullScreen(explicitFullScreenMode: true)
+        } else {
+          toggleFullScreen(explicitFullScreenMode: model.isPortraitVideo)
+        }
+      }
+    }
+    .persistentSystemOverlays(isFullScreen ? .hidden : .automatic)
+    .defersSystemGestures(on: isFullScreen ? .all : [])
+    .ignoresSafeArea(.keyboard, edges: .bottom)
+  }
+}
+
+@available(iOS 16.0, *)
+extension MediaContentView {
+  /// A media scrubber that controls the current player
+  ///
+  /// This is a separate view since it is constantly updating the current time, so its better to
+  /// keep that state isolated and only update this View when time changes instead of the entire
+  /// PlaybackControlsView
+  private struct PlaybackScrubber: View {
+    @ObservedObject var model: PlayerModel
+
+    @State private var currentTime: TimeInterval = 0
+    @State private var isScrubbing: Bool = false
+    @State private var resumePlayingAfterScrub: Bool = false
+
+    var body: some View {
+      MediaScrubber(
+        currentTime: Binding(
+          get: {
+            if isScrubbing {
+              return model.currentTime
+            } else {
+              return self.currentTime
+            }
+          },
+          set: { newValue in
+            Task { await model.seek(to: newValue, accurately: true) }
+          }
+        ),
+        duration: model.duration,
+        isScrubbing: $isScrubbing
+      )
+      .tint(Color(braveSystemName: .iconInteractive))
+      .onChange(of: isScrubbing) { newValue in
+        if newValue {
+          resumePlayingAfterScrub = model.isPlaying
+          model.pause()
+        } else {
+          if resumePlayingAfterScrub {
+            model.play()
+          }
+        }
+      }
+      .task(priority: .low) {
+        self.currentTime = model.currentTime
+        for await currentTime in model.currentTimeStream {
+          self.currentTime = currentTime
+        }
+      }
+    }
+  }
+  struct PlaybackControlsView: View {
+    @ObservedObject var model: PlayerModel
+    var selectedItemTitle: String
+
+    @Environment(\.toggleFullScreen) private var toggleFullScreen
+
+    var body: some View {
+      VStack(spacing: 28) {
+        HStack(alignment: .firstTextBaseline) {
+          Text(selectedItemTitle)
+            .foregroundStyle(Color(braveSystemName: .textPrimary))
+            .font(.headline)
+            .frame(maxWidth: .infinity, alignment: .leading)
+            .multilineTextAlignment(.leading)
+            .lineLimit(1)
+          if model.allowsExternalPlayback {
+            RoutePickerView()
+          }
+        }
+        PlaybackScrubber(model: model)
+        VStack(spacing: 28) {
+          HStack {
+            Toggle(isOn: $model.isShuffleEnabled) {
+              if model.isShuffleEnabled {
+                Image(braveSystemName: "leo.shuffle.toggle-on")
+                  .transition(.opacity.animation(.linear(duration: 0.1)))
```

Comment:
suggestion: saw this transition in many places, it may be a good idea to add this into an extension like playButtonTransition